test/common/upstream/health_check_fuzz.cc

```diff
@@ -112,34 +107,161 @@ void HealthCheckFuzz::triggerTimeoutTimer(bool last_action) {
   }
 }
 
-void HealthCheckFuzz::raiseEvent(const test::common::upstream::RaiseEvent& event,
-                                 bool last_action) {
-  Network::ConnectionEvent eventType;
+void HttpHealthCheckFuzz::raiseEvent(const Network::ConnectionEvent& event_type, bool last_action) {
+  test_sessions_[0]->client_connection_->raiseEvent(event_type);
+  if (!last_action && event_type != Network::ConnectionEvent::Connected) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream from close event");
+    triggerIntervalTimer(
+        true); // Interval timer is guaranteed to be enabled from a close event - calls
+               // onResetStream which handles failure, turning interval timer on and timeout off
+  }
+}
+
+void TcpHealthCheckFuzz::allocTcpHealthCheckerFromProto(
+    const envoy::config::core::v3::HealthCheck& config) {
+  health_checker_ = std::make_shared<TcpHealthCheckerImpl>(
+      *cluster_, config, dispatcher_, runtime_, random_,
+      HealthCheckEventLoggerPtr(event_logger_storage_.release()));
+  ENVOY_LOG_MISC(trace, "Created Tcp Health Checker");
+}
+
+void TcpHealthCheckFuzz::initialize(test::common::upstream::HealthCheckTestCase input) {
+  allocTcpHealthCheckerFromProto(input.health_check_config());
+  cluster_->prioritySet().getMockHostSet(0)->hosts_ = {
+      makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  expectSessionCreate();
+  expectClientCreate();
+  health_checker_->start();
+  if (input.health_check_config().has_reuse_connection()) {
+    reuse_connection_ = input.health_check_config().reuse_connection().value();
+  }
+  // The Receive proto message has a validation that if there is a receive field, the text field, a
+  // string representing the hex encoded payload has a least one byte.
+  if (input.health_check_config().tcp_health_check().receive_size() != 0) {
+    ENVOY_LOG_MISC(trace, "Health checker has a non empty response");
```

Comment:
WOW Nice!!! This is great that you figured this out.
Is this really a non-empty response? I think what `receive_size() == 0` is measuring is if we are matching on the incoming data versus just testing if we can connect. Like "Health checker is expecting a non-empty response to match" or the flip "Health Checker is only testing to connect"

Reply:
Yeah this is. A non empty response simply means that it expects bytes. If this field wasnt there, this would be an empty response, so if the list has size, it means it has a payload message with a string of at least one byte: https://github.com/envoyproxy/envoy/blob/36a59053810f7b919f4313dd90106bc68f713f4a/api/envoy/config/core/v3/health_check.proto#L70. This gets piped into receive bytes in the health checker proto, meaning not empty and expecting a response.

Reply:
I think what this is saying is that the HC config can set text payloads _to match_ the incoming responses from the endpoints.

Like the health checker has a non empty matcher, it expect the end point to have a non empty response (the HC doesn't have responses)

Reply:
But it is totally a semantic thing I'm responding about :)

Reply:
Ohhhhh....I just read my log and youre right that isnt the correct wording let me fix. Thought you were asking about somehting different.

Reply:
Changed to "Health Checker is only testing to connect"